### PR TITLE
Ability to extract keyframes only from a video

### PR DESCRIPTION
**Pull request by mattphotonman**
This is a very fast way to get a sample of frames from a video if you don't care about the exact index of the frames.

### CODE REVIEW

File: eta/core/video.py [hunk 1]
```diff
@@ -1625,6 +1654,7 @@ def __init__(
             self,
             inpath,
             frames=None,
+            keyframes_only=False,
```
Comment (brimoor) [inline eta/core/video.py]: I renamed this to `keyframes_only` for consistency with the `frames` argument (its not `in_frames`).

Note also that I was bold and moved this kwarg up in the list. You had it at the end. We are diligent about using `key=value` syntax whenever using keyword arguments, so this is generally safe to do.

File: eta/core/video.py [hunk 2]
```diff
@@ -1639,6 +1669,9 @@ def __init__(
             inpath: path to the input video. Passed directly to a VideoReader
             frames: an optional range(s) of frames to process. This argument
                 is passed directly to VideoReader
+            keyframes_only: whether to only extract keyframes when reading the
```
Comment (brimoor) [inline eta/core/video.py]: I clarified that you can use `frames` and `keyframes_only` together, which is cool!

File: eta/core/video.py [hunk 3]
```diff
@@ -1973,6 +2020,11 @@ def _grab(self):
             return False
 
     def _retrieve(self):
+        # Stop when ffmpeg returns empty bits, meaning it has gone past the end
+        # of the video
+        if not self._raw_frame:
```
Comment (brimoor) [inline eta/core/video.py]: thanks for adding this too

**Comment by mattphotonman** [comment]
> LGTM. Tested and seems to work. Am I correct in assuming that one can use `keyframes_only` and `frames` together to, for example, extract the first and third keyframes?
> 
> I tested and it appears so. Cool!

Yeah, that's right.

**Pull request merged**